modules/core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepositoryDescriptorRegistry.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ecm.core.storage.mongodb;
+
+import static org.nuxeo.ecm.core.storage.dbs.DBSRepositoryDescriptorRegistry.COMPONENT_NAME;
+import static org.nuxeo.ecm.core.storage.dbs.DBSRepositoryDescriptorRegistry.POINT_NAME;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.common.xmap.Context;
+import org.nuxeo.common.xmap.XAnnotatedObject;
+import org.nuxeo.common.xmap.registry.MapRegistry;
+import org.nuxeo.common.xmap.registry.Registry;
+import org.nuxeo.ecm.core.storage.dbs.DBSRepositoryContributor;
+import org.nuxeo.ecm.core.storage.dbs.DBSRepositoryDescriptorRegistry;
+import org.nuxeo.runtime.RuntimeMessage;
+import org.nuxeo.runtime.RuntimeMessage.Level;
+import org.nuxeo.runtime.RuntimeMessage.Source;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.logging.DeprecationLogger;
+import org.nuxeo.runtime.mongodb.MongoDBComponent;
+import org.w3c.dom.Element;
+
+/**
+ * Registry for {@link MongoDBRepositoryDescriptor}, forwarding to {@link DBSRepositoryDescriptorRegistry}.
+ * <p>
+ * Also handles custom merge.
+ *
+ * @since 11.5
+ */
+public class MongoDBRepositoryDescriptorRegistry extends MapRegistry {
```

Comment:
That's what I did in my last changes indeed, sorry for the change during your review. I also moved the class inside the component class that was otherwise now empty (and lonely and bored, now it's better I hope).